File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/presentproof/common/Predicate.java
package com.evernym.verity.sdk.protocols.presentproof.common;

import com.evernym.verity.sdk.utils.AsJsonObject;
import org.json.JSONObject;

import static com.evernym.verity.sdk.utils.JsonUtil.makeArray;

/**
 * A holder for an predicate based restrictions
 */
public class Predicate implements AsJsonObject  {

    /**
     * Constructs the Predicate object with the given attribute name, type, value and given restrictions

     * @param name the attribute name
     * @param type the type only is supported "<=", "<", ">=" or ">"
     * @param value the value the given attribute must satisfy the type
     * @param restrictions the restrictions for requested presentation for this predicate
     */
    public Predicate(String name, String type, int value, Restriction... restrictions) {
        this.data = new JSONObject()
                .put("name", name)
                .put("p_type", type)
                .put("p_value", value)
                .put("restrictions", makeArray(restrictions));
    }

    final JSONObject data;

    /**
     * Convert this object to a JSON object
     *
     * @return this object as a JSON object
     */
    @Override
    public JSONObject toJson() {
        return data;
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/PresentProof/Common/Predicate.cs
using System.Json;
using VeritySDK.Utils;

namespace VeritySDK.Protocols.PresentProof
{
    /// <summary>
    /// A holder for an predicate based restrictions
    /// </summary>
    public class Predicate : AsJsonObject
    {
        JsonObject data;

        /// <summary>
        /// Constructs the Predicate object with the given attribute name, value and given restrictions 
        /// </summary>
        /// <param name="name">the attribute name</param>
        /// <param name="value">the value the given attribute must be greater than</param>
        /// <param name="restrictions">the restrictions for requested presentation for this predicate</param>
        public Predicate(string name, int value, params Restriction[] restrictions)
        {
            this.data = new JsonObject();
            data.Add("name", name);
            data.Add("p_type", ">=");
            data.Add("p_value", value);
            data.Add("restrictions", JsonUtil.makeArray(restrictions));
        }

        /// <summary>
        /// Convert this object to a JSON object 
        /// </summary>
        /// <returns>this object as a JSON object</returns>
        public JsonObject toJson()
        {
            return data;
        }
    }
}